lib/fog/orchestration/util/recursive_hot_file_loader.rb

```diff
@@ -0,0 +1,232 @@
+require 'yaml'
+require 'open-uri'
+require 'objspace'
+require 'fog/core'
+require 'set'
+
+# Ruby URI is not round-trip safe when schema == file
+#  eg. URI("file:///a.out").to_s != file:///a.out"
+module URI
```

Comment:
@bzwei  I need to extend the URI module to fix round-trip safety. Is that fine? Should I put that class elsewhere?